eng/common/scripts/Detect-Api-Changes.ps1

```diff
@@ -0,0 +1,105 @@
+  [string] $APIViewUri = "https://apiview.dev/PullRequest/DetectApiChanges"
+)
+
+# Submit API review request and return status whether current revision is approved or pending or failed to create review
+function Submit-Request($filePath)
+{
+    $repoName = $RepoFullName
+    if (!$repoName) {
+        $repoName = "azure/azure-sdk-for-$Language"
+    }
+    $query = [System.Web.HttpUtility]::ParseQueryString('')
+    $query.Add('artifactName', $ArtifactName)
+    $query.Add('buildId', $BuildId)
+    $query.Add('filePath', $filePath)
+    $query.Add('commitSha', $CommitSha)
+    $query.Add('repoName', $repoName)
+    $query.Add('pullRequestNumber', $PullRequestNumber)
+    $uri = [System.UriBuilder]$APIViewUri
+    $uri.query = $query.toString()
+    Write-Host "Request URI: $($uri.Uri.OriginalString)"
+    try
+    {
+        $Response = Invoke-WebRequest -Method 'GET' -Uri $uri.Uri
+        $StatusCode = $Response.StatusCode
+    }
+    catch
+    {
+        Write-Host "Error $StatusCode - Exception details: $($_.Exception.Response)"
+        $StatusCode = $_.Exception.Response.StatusCode
+    }
+
+    return $StatusCode
+}
+
+function Should-Process-Package($pkgPath, $packageName)
+{
+    $pkg = Split-Path -Leaf $pkgPath
+    $configFileDir = Join-Path -Path $ArtifactPath "PackageInfo"
+    $pkgPropPath = Join-Path -Path $configFileDir "$packageName.json"
+    if (-Not (Test-Path $pkgPropPath))
+    {
+        Write-Host " Package property file path $($pkgPropPath) is invalid."
+        return $False
+    }
+    # Get package info from json file created before updating version to daily dev
+    $pkgInfo = Get-Content $pkgPropPath | ConvertFrom-Json
+    Write-Host "SDK Type: $($pkgInfo.SdkType)"
+    return ($pkgInfo.SdkType -eq "client" -and $pkgInfo.IsNewSdk)
+}
+
+function Log-Input-Params()
+{
+    Write-Host "Artifact Path: $($ArtifactPath)"
+    Write-Host "Artifact Name: $($ArtifactName)"
+    Write-Host "PullRequest Number: $($PullRequestNumber)"
+    Write-Host "BuildId: $($BuildId)"
+    Write-Host "Language: $($Language)"
+    Write-Host "Commit SHA: $($CommitSha)"
+    Write-Host "Repo Name: $($RepoFullName)"
+}
+
+. (Join-Path $PSScriptRoot common.ps1)
+Log-Input-Params
+
+if (!($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn")))
+{
+    Write-Host "The function for 'FindArtifactForApiReviewFn' was not found.`
+    Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+    See https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/common_engsys.md#code-structure"
+    exit 1
+}
+
+$artifacts = $ArtifactList.Split(",")
+foreach ($pkgName in $artifacts)
+{
+    Write-Host "Processing $pkgName"
+    $packages = &$FindArtifactForApiReviewFn $ArtifactPath $pkgName
+    if ($packages)
+    {
+        $pkgPath = $packages.Values[0]
+        if (Should-Process-Package -pkgPath $pkgPath -packageName $pkgName)
```

Comment:
Perhaps instead of having the Should-Process-Package you can let the languages only return packages that it should process which I believe is what is already happening.

Reply:
We refer the value in Package Info json artifact to identify SDK type. Most of the code in this method is to parse this package info file. As mentioned above, I will change this to refer a property that's set in package info to decide whether API changes are tracked and approved or not for a given package.